release/create.sh

```diff
@@ -3,9 +3,14 @@
 set -euo pipefail
 
 ARTIFACT_CLI_VERSION="v0.6.3"
-WHEN_CLI_VERSION="v1.0.5"
 SPC_CLI_VERSION="v1.10.1"
 TEST_RESULTS_CLI_VERSION="v0.7.0"
+WHEN_CLI_VERSION="v1.2.0"
+# we include multiple when binaries for all suported Erlang versions
+# and configure the correct one based on Erlang version in the VM where toolbox is installed
+WHEN_BINARY_VERSION_1="when_otp_24"
+WHEN_BINARY_VERSION_2="when_otp_25"
+WHEN_BINARY_VERSION_3="when_otp_26"
```

Comment:
@DamjanBecirovic 
Should you add for Erlang 27 as well?

Reply:
We need support for Erlang 24 because it is the default version for ubuntu2004 images and Erlang 25 because of the ubuntu2204 image. The Erlang 27 is currently not used as the default anywhere.

Based on [this](https://hexdocs.pm/elixir/1.17.2/compatibility-and-deprecations.html#between-elixir-and-erlang-otp), the only version of Elixir that supports Erlang 27 is not compatible with Erlang 24.
Due to that, we would need to have two versions of the when CLI source code, one for Elixir 16 and another for Elixir 17, which would complicate things a lot. 

We can change this in the future, once we drop support for ubuntu2004 and introduce ubuntu2404 or ubuntu2604 that will use Erlang 27 as the default version of Erlang.

Reply:
@DamjanBecirovic 

`ubuntu2204` comes with 27.0 as default: 
(I realized I missed to update this in the docs, it will be fixed)

```
semaphore@semaphore-vm:~$ lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 22.04.4 LTS
Release:	22.04
Codename:	jammy

semaphore@semaphore-vm:~$ cat .kerl/otp_installations
27.0 /home/semaphore/.kerl/installs/27.0
```

We also have `ubuntu2404` in the "release pending" state which comes with [27.0 as default](https://docs.semaphoreci.com/ci-cd-environment/ubuntu-24.04-image/#erlang-and-elixir).

Our internal matrix for Erlang-Elixir dependency is [here](https://github.com/semaphoreci/toolbox/blob/9e950bd72672095776a5f3950106fcb22b5f2852/sem-install#L73).
We tried to simplify it since Elixir requires a specific Erlang which was used to build it. Otherwise, I have to rebuild them every time and release a new toolbox with that matrix change.

Erlang 27.1 is out as well, so I will have to decide what to do (at least) with Elixir 1.7.x and our internal matrix (which will require rebuilds of Elixir). 

If you have any suggestions let me know.